editoast/src/modelsv2/projects.rs

```diff
@@ -0,0 +1,253 @@
+use crate::error::Result;
+use crate::modelsv2::{Changeset, DeleteStatic, Document, Model, Retrieve, Row, Save, Update};
+use crate::views::pagination::{Paginate, PaginatedResponse};
+use crate::views::projects::ProjectError;
+use crate::DbPool;
+use actix_web::web::Data;
+use async_trait::async_trait;
+use chrono::{NaiveDateTime, Utc};
+use diesel::{sql_query, ExpressionMethods, QueryDsl};
+use diesel_async::{AsyncPgConnection as PgConnection, RunQueryDsl};
+use editoast_derive::ModelV2;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::models::List;
+
+crate::schemas! {
+    Ordering,
+    Project,
+    Tags,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, ModelV2, ToSchema, PartialEq)]
+#[model(table = crate::tables::project)]
+pub struct Project {
+    pub id: i64,
+    pub name: String,
+    pub objectives: String,
+    pub description: String,
+    pub funders: String,
+    pub budget: i32,
+    pub creation_date: NaiveDateTime,
+    pub last_modification: NaiveDateTime,
+    #[model(remote = "Vec<Option<String>>")]
+    pub tags: Tags,
+    #[model(column = "image_id")]
+    pub image: Option<i64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default, ToSchema, PartialEq)]
+pub struct Tags(Vec<String>);
+
+impl Tags {
+    pub fn new(value: Vec<String>) -> Self {
+        Self(value)
+    }
+}
+
+impl From<Vec<Option<String>>> for Tags {
+    fn from(value: Vec<Option<String>>) -> Self {
+        Self(value.into_iter().flatten().collect())
+    }
+}
+
+impl From<Tags> for Vec<Option<String>> {
+    fn from(value: Tags) -> Self {
+        value.0.into_iter().map(Some).collect()
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Default, ToSchema)]
+pub enum Ordering {
+    NameAsc,
+    NameDesc,
+    CreationDateAsc,
+    CreationDateDesc,
+    #[default]
+    LastModifiedDesc,
+    LastModifiedAsc,
+}
+
+impl Ordering {
+    pub fn to_sql(&self) -> &str {
+        match *self {
+            Ordering::NameAsc => "LOWER(t.name) ASC",
+            Ordering::NameDesc => " LOWER(t.name) DESC",
+            Ordering::CreationDateAsc => "creation_date",
+            Ordering::CreationDateDesc => "creation_date DESC",
+            Ordering::LastModifiedAsc => "last_modification",
+            Ordering::LastModifiedDesc => "last_modification DESC",
+        }
+    }
+}
+
+impl Project {
+    /// This function takes a filled project and update to now the last_modification field
+    pub async fn update_last_modified(self, db_pool: Data<DbPool>) -> Result<()> {
```

Comment:
```suggestion
    pub async fn update_last_modified(&mut self, db_pool: Data<DbPool>) -> Result<()> {
```

That way you won't have to `.clone()` the project to update it in views::study.

Reply:
Done.